mac/Keyman4MacIM/Keyman4MacIM/KeySender.m

```diff
@@ -90,23 +63,20 @@ - (void)postKeyboardEventWithSource: (CGEventSourceRef)source code:(CGKeyCode) v
 - (void)sendKeymanKeyCodeForEvent:(NSEvent *)event {
   [self.appDelegate logDebugMessage:@"KeySender sendKeymanKeyCodeForEvent"];
   
-  [self sendKeyDown:kKeymanEventKeyCode forSourceEvent:event includeKeyUp:NO];
-  
+  ProcessSerialNumber psn;
+
   // Returns the frontmost app, which is the app that receives key events.
   NSRunningApplication *app = NSWorkspace.sharedWorkspace.frontmostApplication;
   pid_t processId = app.processIdentifier;
   NSString *bundleId = app.bundleIdentifier;
+  GetProcessForPID(processId, &psn);
   
   [self.appDelegate logDebugMessage:@"sendKeymanKeyCodeForEvent keyCode %lu to app %@ with pid %d", (unsigned long)kKeymanEventKeyCode, bundleId, processId];
   
-  CGEventRef cgevent = [event CGEvent];
-  
-  // use source from event to generate new event
-  CGEventSourceRef source = CGEventCreateSourceFromEvent(cgevent);
-  CGEventRef keyDownEvent = CGEventCreateKeyboardEvent(source, kKeymanEventKeyCode, true);
-  
-  // TODO: add version check
-  CGEventPostToPid(processId, keyDownEvent);
+  // use nil as source, as this generated event is not directly tied to the originating event
+  CGEventRef keyDownEvent = CGEventCreateKeyboardEvent(nil, kKeymanEventKeyCode, true);
+
+  CGEventPostToPSN(&psn, keyDownEvent);
```

Comment:
Why has this been changed from `CGEventPostToPid`?

Reply:
CGEventPostToPid is only supported with macOS 10.11 and later, and we still support macOS 10.10 in 17. CGEventPostToPSN, though deprecated, still works ok with Sonoma. I prefer CGEventPostToPid, and it is a cleaner interface, but I'll wait until 18 when we move to the Xcode 15 and stop supporting 10.10.